docker/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+#
+# This file is part of the IoT Agent for the Ultralight 2.0 protocol (IOTAUL) component
+#
+# IOTAUL is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# IOTAUL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with IOTAUL.
+# If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by the GNU Affero General Public License
+# please contact with: [daniel.moranjimenez@telefonica.com]
+
+ARG NODE_VERSION=8.12.0-slim
+FROM node:${NODE_VERSION}
+ARG GITHUB_ACCOUNT=telefonicaid
+ARG GITHUB_REPOSITORY=iotagent-ul
+ARG DOWNLOAD_TYPE=latest
+
+ENV GITHUB_ACCOUNT=${GITHUB_ACCOUNT}
+ENV GITHUB_REPOSITORY=${GITHUB_REPOSITORY}
+ENV DOWNLOAD_TYPE=${DOWNLOAD_TYPE}
+
```

Comment:
What't the purpose of having a parallel set of ARG and ENV? I understand it for some good reason :) but at a glance it a bit redundant and add complexity.

Could you elaborate, please?

Reply:
Yes it's kind of ugly, but it is standard practice to obtain build time environment variables. 

`ARG` is only available on build, so setting it as an `ENV` enables it to stick around when the container is running (hence you can set the value with a `--build-arg` and then read it from the running container (so `docker exec <my-image-name> env` should return information about the GitHub repo the image was built against.

* https://vsupalov.com/docker-build-time-env-values/
* https://vsupalov.com/docker-arg-env-variable-guide/

Reply:
+1 to https://github.com/telefonicaid/iotagent-ul/pull/310/files#r236232373

Reply:
Fixed - Explanation added 03e74ec